src/features/datamodel/DataModelsProvider.tsx

```diff
@@ -198,17 +192,10 @@ function DataModelsLoader() {
 }
 
 function BlockUntilLoaded({ children }: PropsWithChildren) {
-  const {
-    allDataTypes,
-    writableDataTypes,
-    initialData,
-    initialValidations,
-    schemas,
-    expressionValidationConfigs,
-    error,
-  } = useSelector((state) => state);
+  const { allDataTypes, writableDataTypes, initialData, schemas, expressionValidationConfigs, error } = useSelector(
+    (state) => state,
+  );
```

Comment:
It wasn't any better before, but selecting all state means this provider will re-render all children when any state changes - have you checked if this has a performance impact?

Reply:
A trick to make this perform better is to check the things you need to check _inside_ the selector, and then just return a boolean 'isReady' from the selector.

Reply:
The state here is only really updated as things are loading, and should not get any further updates until loading the next process step, so I dont think this has any performance implications when filling out a form